dvc/repo/plots/template.py

```diff
@@ -533,15 +542,38 @@ def get_template(self, template_name: str):
     def __init__(self, dvc_dir):
         self.dvc_dir = dvc_dir
 
-    def init(self):
+    def init(
+        self, output: Optional[str] = None, targets: Optional[List] = None
+    ):
         from dvc.utils.fs import makedirs
 
-        makedirs(self.templates_dir, exist_ok=True)
-        for t in self.TEMPLATES:
-            self._dump(t())
+        output = output or self.templates_dir
+
+        makedirs(output, exist_ok=True)
+
+        if targets:
+            templates = [
+                template
+                for template in self.TEMPLATES
+                if template.DEFAULT_NAME in targets
+            ]
+        else:
+            templates = self.TEMPLATES
+
+        for template in templates:
+            self._dump(template(), output)
+
+    def _dump(self, template: Template, output: str):
+        path = os.path.join(output, template.filename)
+
+        if os.path.exists(path):
+            with open(path, "r", encoding="utf-8") as fd:
+                content = fd.read()
+            if content != template.content:
```

Comment:
Probably is simpler this way but if this is False there is no need for the re-write below.

Reply:
fixed